corehq/ex-submodules/dimagi/utils/decorators/profile.py

```diff
@@ -21,38 +26,65 @@
 # Source: http://code.djangoproject.com/wiki/ProfilingDjango
 
 
-def profile(log_file):
+def profile(log_file, probability=1, limit=None):
```

Comment:
0 seems like a safer default to me. Also limit seems like it should be 1 so that we can't accidentally fill our disk with a bunch of profiles

Reply:
yeah those seem like better defaults for production. 

probably over thinking this, but if we wanted to keep the dev default we could make the function args `ellipsis` or whatever, and then set the defaults inside the function to `probability=1, limit=None` if `settings.DEBUG=True` and `probability=0, limit=1` if `settings.DEBUG` is `False` (or not present).

but also typing `@profile('name', 1, None)` during dev seems totally fine.